connect_ext_test.go

```diff
@@ -387,6 +388,44 @@ func TestServer(t *testing.T) {
 	})
 }
 
+func TestConcurrentStreams(t *testing.T) {
+	if testing.Short() {
+		t.Skipf("skipping %s test in short mode", t.Name())
+	}
+	t.Parallel()
+	mux := http.NewServeMux()
+	mux.Handle(pingv1connect.NewPingServiceHandler(pingServer{}))
+	server := httptest.NewUnstartedServer(mux)
+	server.EnableHTTP2 = true
+	server.StartTLS()
+	t.Cleanup(server.Close)
+	var done sync.WaitGroup
+	var startWg sync.WaitGroup
+	startWg.Add(1)
+	for i := 0; i < 100; i++ {
+		done.Add(1)
+		go func() {
+			defer done.Done()
+			client := pingv1connect.NewPingServiceClient(server.Client(), server.URL)
+			var total int64
+			sum := client.CumSum(context.Background())
+			startWg.Wait()
+			for i := 0; i < 100; i++ {
+				num := rand.Int63n(1000)
+				total += num
+				assert.Nil(t, sum.Send(&pingv1.CumSumRequest{Number: num}))
+				resp, err := sum.Receive()
+				assert.Nil(t, err)
+				assert.Equal(t, total, resp.Sum)
```

Comment:
I'm fairly sure that `assert` uses methods on `testing.T` that aren't safe to call concurrently. You may want to double-check me on that, though :)

If I'm right about that, I think we could get away with a `chan error` and some non-blocking writes in the client goroutines. After all the client goroutines shut down, we can try reading from the channel and fail the test if we find an error. (I don't think we want to serialize the test by introducing locks.)

Reply:
Replaced calls to internal `assert.*` with `t.Error*` calls which appear safe to call from multiple goroutines.